src/wallet/bdb.h

```diff
@@ -303,8 +304,9 @@ class BerkeleyBatch
         return HasKey(ssKey);
     }
 
-    Dbc* GetCursor();
-    int ReadAtCursor(Dbc* pcursor, CDataStream& ssKey, CDataStream& ssValue);
+    bool CreateCursor();
+    bool ReadAtCursor(CDataStream& ssKey, CDataStream& ssValue, bool& complete);
+    void CloseCursor();
```

Comment:
In commit "walletdb: Handle cursor internally" (84bc5c1705691f4ecdfb8bcc90eb1bc4970281dc)

Would it make sense to call CloseCursor in the BerkeleyBatch destructor? It seems like it could guard against memory leaks, and maybe avoid the need to add CloseCursor calls some of the places where they are added.

Reply:
Done (I put it in `Close` which is called by the desctructor).